### ISSUE TITLE
sctool progress fails when checking the progress of a restore task: `invalid memory address or nil pointer dereference`

### ISSUE THREAD

**Issue by ShlomiBalalis** [scylladb/scylla-manager#3289]
Client version: 3.1-dev-0.20230111.71ee737dafe-SNAPSHOT-71ee737dafe
Server version: 3.1-dev-0.20230111.71ee737dafe-SNAPSHOT-71ee737dafe

In this scenario I run a backup task to completion and trunate the table that was backed up. Afterwards we add a new node to the cluster and run a restore, like so:
```
sctool restore -c 9122dc14-225b-4839-9941-0f88ff1dfe9e --restore-tables --location s3:backup-bucket --snapshot-tag sm_20230116125214UTC

restore/c31ed335-d2c8-4b85-b186-4a3d575b2d57
```
But when I try to examine the progress of the task, the command fails:
```
sctool progress restore/c31ed335-d2c8-4b85-b186-4a3d575b2d57 -c 9122dc14-225b-4839-9941-0f88ff1dfe9e
Run:		e2a9fcff-959c-11ed-a013-f4ee08c9cc47
Status:		Error: template: restore_progress:3:12: executing "restore_progress" at <status>: error calling status: runtime error: invalid memory address or nil pointer dereference
```

The issue seems to be consistent in this scenario and in this scenario alone.

Logs:
[scylla-manager.log](https://github.com/scylladb/scylla-manager/files/10427506/scylla-manager.log)
Agent logs:
* [manager_agent_1.log](https://github.com/scylladb/scylla-manager/files/10427521/manager_agent_1.log)
* [manager_agent_2.log](https://github.com/scylladb/scylla-manager/files/10427528/manager_agent_2.log)
* [manager_agent_3.log](https://github.com/scylladb/scylla-manager/files/10427531/manager_agent_3.log) (The node that was added)

**Comment by Michal-Leszczynski** [comment]
I believe that the following thing happened:
- SM got restore target (preparation before restore and before inserting restore run into DB)
- SM was queried for restore progress
- SM should return 404 error because it couldn't find restore run in DB, but in this scenario SM prefers to return empty progress information instead of error and that's the place where I forgot to insert empty restore progress instead of `nil`
- SM crashed because of this bug

So It's not about really about adding new node to the cluster, but rather about asking for restore progress before restore run could be inserted into DB.
Anyways, it's a bug and I already created fix for it (#3293) but I still need to verify if this for sure fixes that issue.

**Comment by Michal-Leszczynski** [comment]
I verified that querying restore progress right after restore was scheduled returns this error.
That means that #3293 indeed fixes this issue.

**Comment by ShlomiBalalis** [comment]
Client version: 3.1.0-rc0-0.20230307.e99afd24
Server version: 3.1.0-rc0-0.20230307.e99afd24

The issue is recreated consistently when attempting to restore while one of the nodes is out of disk space. We first fill node 2 until it runs out of disk space, and then create a restore task:
```
< t:2023-03-09 19:26:51,306 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "sudo sctool restore -c 47d3ca56-94cc-4d2c-8189-6e2147db3e95 --restore-tables --location s3:manager-backup-tests-us-east-1  --snapshot-tag sm_20230309191653UTC" finished with status 0

< t:2023-03-09 19:26:51,307 f:cli.py          l:1110 c:sdcm.mgmt.cli        p:DEBUG > sctool output: restore/c2fc1549-a869-4b14-bd73-1a970d0ad40c
```
But as soon as we try to check the progress of the restore, the progress command fails with the same error:
```
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > Command: 'sudo sctool  -c 47d3ca56-94cc-4d2c-8189-6e2147db3e95 progress restore/c2fc1549-a869-4b14-bd73-1a970d0ad40c'
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > 
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > Exit code: 1
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > 
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > Stdout:
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > 
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > Run:              57643a59-beb0-11ed-bafc-0294c34c4779
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > Status:           
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > 
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > Stderr:
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > 
< t:2023-03-09 19:26:54,589 f:wait.py         l:49   c:sdcm.wait            p:DEBUG > Error: template: restore_progress:3:12: executing "restore_progress" at <status>: error calling status: runtime error: invalid memory address or nil pointer dereference
```
We try to check the progress of the task for over 10 minutes afterwards, with the same results.

Logs:
[db-cluster - https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/db-cluster-a981de58.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/db-cluster-a981de58.tar.gz)
[loader-set - https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/loader-set-a981de58.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/loader-set-a981de58.tar.gz)
[monitor-set - https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/monitor-set-a981de58.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/monitor-set-a981de58.tar.gz)
[sct - https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/sct-runner-a981de58.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a981de58-9c8d-4f4c-a391-743624d06ae4/20230309_205843/sct-runner-a981de58.tar.gz)

**Comment by Michal-Leszczynski** [comment]
SM logs shows that all used nodes have about 96% of free space, so it shouldn't be connected to that.
We can see that an error from #3326 was replicated here, because SM got stuck on restoring an empty table.

But most importantly, I think that this issue was indeed solved by #3293, but unfortunately, the commit that fixes that was pushed only to master and not to the 3.1 branch.

**Comment by karol-kokoszka** [comment]
Branch 3.1 was updated recently. 
These tests must be available in jenkins so that we can run them.
Please reopen if you see that it still appears.
